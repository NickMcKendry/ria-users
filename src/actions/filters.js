
export const setTextFilter = (filter = '') => ({
  type: 'SET_TEXT_FILTER',
  filter
});

export const sortByFirstName = () => ({
  type: 'SORT_BY_FIRST_NAME'
});

export const sortByLastName = () => ({
  type: 'SORT_BY_LAST_NAME'
});
