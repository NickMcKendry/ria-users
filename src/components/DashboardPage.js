import React from 'react';
import ReactDOM from 'react-dom';

const DashboardPage = () => (
  <div>
    <p>DashboardPage content</p>
  </div>
);

export default DashboardPage;
