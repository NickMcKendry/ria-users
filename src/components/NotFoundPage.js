import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <div>
    <h3>404 - <Link to="/">Go Home</Link></h3>
  </div>
);

export default NotFoundPage;
