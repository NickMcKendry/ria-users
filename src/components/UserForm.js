import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';


export default class UserForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      firstName: props.user ? props.user.firstName : '',
      lastName: props.user ? props.user.lastName : '',
      address: props.user ? props.expense.address : '',
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calendarFocused: false,
      error: ''
    };
  };


  onFirstNameChange = (e) => {
    const firstName = e.target.value;
    this.setState(() => ({ firstName }));
  };

  onLastNameChange = (e) => {
    const lastName = e.target.value;
    this.setState(() => ({ lastName }));
  };

  onAddressChange = (e) => {
    const address = e.target.value;
    this.setState(() => ({ address }))
  };

  onDateChange = (createdAt) => {
    if(createdAt){
      this.setState(() => ({ createdAt }));
    }
  };

  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }))
  };

  onSubmit = ((e) => {
    e.preventDefault();

    if (!this.state.firstName || !this.state.lastName || !this.state.address) {
      this.setState(() => ({ error: 'Please provide a firstName, lastName and address!' }));
    } else {
      this.setState(() => ({ error: '' }));
      this.props.onSubmit({
        firstName: this.state.firstName,
        lastName: this.state.lastName,
        createdAt: this.state.createdAt.valueOf(),
        address: this.state.address
      })
    }
  })

  render() {
    return (
        <form onSubmit={this.onSubmit} className="form">
          { this.state.error && <p className="form__error">{this.state.error}</p> }
          <input
            type="text"
            placeholder="First Name"
            autoFocus
            value={this.state.firstName}
            onChange={this.onFirstNameChange}
            className="text-input"
          />
          <input
            type="text"
            placeholder="Last Name"
            value={this.state.lastName}
            onChange={this.onLastNameChange}
            className="text-input"
          />
          <input
            type="text"
            placeholder="Address"
            value={this.state.address}
            onChange={this.onAddressChange}
            className="text-input"
          />
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
            <div>
              <button className="button">Save User</button>
            </div>

        </form>
    )
  }
};
