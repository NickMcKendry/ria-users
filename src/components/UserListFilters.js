import React from 'react';
import { connect } from 'react-redux';
import {
  setTextFilter,
  sortByFirstName,
  sortByLastName,
} from '../actions/filters';


export class UserListFilters extends React.Component {

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

  onSortChange = (e) => {
    if(e.target.value === 'first-name'){
        this.props.sortByFirstName();
      } else if(e.target.value === 'last-name'){
        this.props.sortByLastName();
      }
    };

  render() {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
              <input
                type="text"
                value={this.props.filters.text}
                onChange={this.onTextChange}
                className="text-input"
                placeholder="Search Users"
              />
          </div>
          <div className="input-group__item">
            <select
              value={this.props.filters.sortBy}
              onChange={this.onSortChange}
              className="select"
            >
              <option value="first-name">First Name</option>
              <option value="last-name">Last Name</option>
            </select>
          </div>
        </div>
      </div>
    )
  }
};

const mapStateToProps = (state) => ({
  filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByFirstName: () => dispatch(sortByFirstName()),
  sortByLastName: () => dispatch(sortByLastName()),
});

export default connect(mapStateToProps, mapDispatchToProps)(UserListFilters);
