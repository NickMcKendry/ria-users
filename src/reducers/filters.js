import moment from 'moment';

const filtersReducerDefaultState = {
  text: '',
  sortBy: 'first-name'
};

export default (state = filtersReducerDefaultState, action) => {
  switch (action.type) {
    case 'SET_TEXT_FILTER' :
      return {
        ...state,
        text: action.filter
      };
    case 'SORT_BY_FIRST_NAME' :
      return {
        ...state,
        sortBy: 'first-name'
      };
    case 'SORT_BY_LAST_NAME' :
      return {
        ...state,
        sortBy: 'last-name'
      };
    default :
      return state;
  };
};
