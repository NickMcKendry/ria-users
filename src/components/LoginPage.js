import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Boilerplate v2</h1>
      <p>App Tagline</p>
      <button onClick={startLogin} className="button">Login With Google</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
