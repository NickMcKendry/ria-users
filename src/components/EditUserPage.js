import React from 'react';
import UserForm from './UserForm';
import { connect } from 'react-redux';
import { startEditUser, startRemoveUser } from '../actions/users';

export class EditUserPage extends React.Component {


  onSubmit = (user) => {
    this.props.editUser(this.props.user.id, user);
    this.props.history.push('/');
  };

  onClick = () => {
    this.props.removeUser({ id: this.props.user.id });
    this.props.history.push('/');
  }

  render() {
    return(
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit user</h1>
          </div>
        </div>
        <div className="content-container">
          <UserForm
            user={this.props.user}
            onSubmit={this.onSubmit}
          />
          <button onClick={this.onClick} className="button button--secondary">Remove User</button>
        </div>
      </div>
    );
  };
};

const mapStateToProps = (state, props) => {
  return {
    user: state.users.find((user) => user.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch, props) => ({
  editUser: (id, user) => dispatch(startEditUser(id, user)),
  removeUser: (id) => dispatch(startRemoveUser(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditUserPage);
