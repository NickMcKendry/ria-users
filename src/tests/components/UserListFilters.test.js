import React from 'react';
import { shallow } from 'enzyme';
import { UserListFilters } from '../../components/UserListFilters';
import { filters, altFilters } from '../fixtures/filters';

let setTextFilter, sortByFirstName, sortByLastName, wrapper;

beforeEach(() => {
  setTextFilter = jest.fn();
  sortByFirstName = jest.fn();
  sortByLastName = jest.fn();
  wrapper = shallow(
    <UserListFilters
      setTextFilter={setTextFilter}
      sortByFirstName={sortByFirstName}
      sortByLastName={sortByLastName}
      filters={filters}
    />
  );
});

test('Should render UserListFilters', () => {
  expect(wrapper).toMatchSnapshot();
});

test('Should render UserListFilters with alt data', () => {
  wrapper.setProps({
    filters: altFilters
  })
  expect(wrapper).toMatchSnapshot();
});

test('Should handle text change', () => {
  const value = altFilters.text
  wrapper.find('input').at(0).simulate('change', {
    target: { value }
  });
  expect(setTextFilter).toHaveBeenLastCalledWith(value)
});

test('Should sort by firstName', () => {
  const value = 'first-name'

  wrapper.find('select').simulate('change', {
    target: { value }
  });
  expect(sortByFirstName).toHaveBeenCalled();
});

test('Should sort by lastName', () => {
  wrapper.find('select').simulate('change', {
    target: { value: 'last-name' }
  });
  expect(sortByLastName).toHaveBeenCalled();
});
