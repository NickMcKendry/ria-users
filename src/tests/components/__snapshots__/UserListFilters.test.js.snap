// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render UserListFilters 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search Users"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group__item"
    >
      <select
        className="select"
        onChange={[Function]}
        value="first-name"
      >
        <option
          value="first-name"
        >
          First Name
        </option>
        <option
          value="last-name"
        >
          Last Name
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`Should render UserListFilters with alt data 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search Users"
        type="text"
        value="Bob"
      />
    </div>
    <div
      className="input-group__item"
    >
      <select
        className="select"
        onChange={[Function]}
        value="last-name"
      >
        <option
          value="first-name"
        >
          First Name
        </option>
        <option
          value="last-name"
        >
          Last Name
        </option>
      </select>
    </div>
  </div>
</div>
`;
